tests/e2e/mgmt_wrkld_domain_isolation_negative.go

```diff
@@ -0,0 +1,485 @@
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Creating pvc")
+		pvclaim1, _, err := createPVCAndQueryVolumeInCNS(ctx, client, namespace, labelsMap, "",
+			diskSize, storageclassZ1, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, _, err = createPVCAndQueryVolumeInCNS(ctx, client, namespace, labelsMap, "",
+			diskSize, storageclassZ2, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating service")
+		_ = CreateService(namespace, client)
+
+		ginkgo.By("Creating statefulset")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, false, nil,
+			false, true, "", "", sharedStorageClass, storagePolicyNameZ2)
+
+		ginkgo.By("Perform wcp service down and at the same time mark zone2 for removal")
+		expectedStatusCodes := []int{status_code_failure}
+		var wg sync.WaitGroup
+		wg.Add(2)
+		go markZoneForRemovalFromWcpNsWithWg(vcRestSessionId, namespace,
+			zone2, expectedStatusCodes, &wg)
+		go stopWcpWithWg(ctx, vcAddress, &wg)
+		isWcpServicestopped = true
+		wg.Wait()
+		defer func() {
+			if isWcpServicestopped {
+				ginkgo.By(fmt.Sprintln("Starting WCP on the vCenter host"))
+				err := invokeVCenterServiceControl(ctx, startOperation, wcpServiceName, vcAddress)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Perform a scaling operation on the StatefulSet, increasing the replica count to 4.")
+		err = performScalingOnStatefulSetAndVerifyPvNodeAffinity(ctx, client,
+			4, 0, sts, true, namespace, allowedTopologies, true, false, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, sts, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create new pvc with requested topology of the zone2 which is marked for removal")
+		pvc3, err := createPvcWithRequestedTopology(ctx, client, namespace, nil, "", storageclassZ2, "", zone2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		err = fpv.WaitForPersistentVolumeClaimPhase(ctx,
+			v1.ClaimPending, client, pvc3.Namespace, pvc3.Name, framework.Poll, time.Minute/2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating a Deployment")
+		dep, err := createDeployment(ctx, client, 1, labelsMap, nil, namespace,
+			[]*v1.PersistentVolumeClaim{pvclaim1}, execRWXCommandPod1, false, busyBoxImageOnGcr)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, err = fdep.GetPodsForDeployment(ctx, client, dep)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		allowedTopologies = setSpecificAllowedTopology(allowedTopologies, topkeyStartIndex, 0,
+			1)
```

Comment:
Would be good if you can give a comment here what is 0 and 1 here
This applies to all the testcases where you have called this method